docs/Spring Boot/5_Step_To_Make_Gradle_Configuration_Extreme_Clean_In_A_Multi-Module_Project.md

```diff
@@ -0,0 +1,216 @@
+        id 'com.google.cloud.tools.jib' version "${jibVersion}"
+    }
+}
+
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.FAIL_ON_PROJECT_REPOS)
+    repositories {
+        mavenCentral()
+        mavenLocal()
+        maven {
+            url '.m2/local/'
+        }
+    }
+}
+
+rootProject.name = 'event-sourcing-order-poc'
+
+include 'modules'
+include 'modules:common'
+findProject(':modules:common')?.name = 'common'
+// ... and other modules settings
+
+include 'order'
+include 'order:command-side'
+findProject(':order:command-side')?.name = 'order-command-side'
+include 'order:event-handler'
+findProject(':order:event-handler')?.name = 'order-event-handler'
+include 'order:query-side'
+findProject(':order:query-side')?.name = 'order-query-side'
+// ... and other sub-project settings
+
+```
+
+### Step 3: Define Allprojects DependencyManagement
+
+All the used libraries should be defined in a `allprojects.dependencyManagement` closure in `build.gradle` of the root module:
+
+```groovy title="./build.gradle"
+import org.springframework.boot.gradle.plugin.SpringBootPlugin
+
+plugins {
+    id 'java'
+    id 'java-library'
+    id 'io.spring.dependency-management'
+    id 'org.springframework.boot' apply false
+    id 'com.google.cloud.tools.jib' apply false
+}
+
+allprojects {
+
+    java {
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
+    }
+
+    apply plugin: 'java'
+    apply plugin: 'io.spring.dependency-management'
+    apply plugin: 'java-library'
+
+    dependencyManagement {
+        imports {
+            mavenBom SpringBootPlugin.BOM_COORDINATES
+            mavenBom "org.springframework.cloud:spring-cloud-dependencies:${springCloudVersion}"
+        }
+        dependencies {
+            dependency "org.springdoc:springdoc-openapi-starter-webmvc-ui:${springdocVersion}"
+            dependency "io.github.openfeign:feign-micrometer:${feignMicrometerVersion}"
+            dependency "org.projectlombok:lombok:${lombokVersion}"
+            dependency "org.wiremock:wiremock:${wiremockVersion}"
+            dependency "com.github.loki4j:loki-logback-appender:${logbackAppenderVersion}"
+        }
+    }
+
+    dependencies {
+        // only declare all-needed dependencies
+        compileOnly "org.projectlombok:lombok:${lombokVersion}"
+        annotationProcessor "org.projectlombok:lombok:${lombokVersion}"
+    }
+
+    test {
+        useJUnitPlatform()
+    }
+
+}
+
+tasks.named("jar") {
+    enabled = false
+}
+```
+
+in the `dependencyManagement` closure, we can first import the BOM of other dependencies project like [spring-boot-dependencies](https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies) and [spring-cloud-dependencies](https://mvnrepository.com/artifact/org.springframework.cloud/spring-cloud-dependencies). Then, we can declare the version of other used libraries.
```

Comment:
📝 **[vale]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[Google.Acronyms] Spell out 'BOM', if it's unfamiliar to the audience.